docs/api/grpc.md

```diff
@@ -4,16 +4,358 @@ sidebar_position: 30
 description: 'Reference documentation for the gRPC API'
 ---
 
-TODO
+This document describes the gRPC API provided by Inspektor Gadget for remote control of gadgets, instance management, and trace data collection.
 
-## api.proto
+## api.proto​
 
-TODO
+The `api.proto` defines the core gRPC services and message types for interacting with Inspektor Gadget.
+
+### Services​
+
+The API provides three main services:
+
+#### BuiltInGadgetManager​
+
+Service for managing built-in gadgets that come pre-packaged with Inspektor Gadget.
```

Comment:
The section about builtin gadgets should include the warning that they are getting deprecated. You can reuse the warning from docs/gadgets/builtin/index.mdx

I am wondering about moving this section to the bottom of the page since the docs should not focus on deprecated topics.